conda_smithy/configure_feedstock.py

```diff
@@ -1114,6 +1114,7 @@ def _azure_specific_setup(jinja_env, forge_config, forge_dir, platform):
         if yum_build_setup:
             forge_config["yum_build_setup"] = yum_build_setup
 
+    forge_config = deepcopy(forge_config)
```

Comment:
@leofang, can you try printing `forge_config here?hgfd

Reply:
```python
{'docker': {'executable': 'nvidia-docker', 'fallback_image': 'condaforge/linux-anvil-comp7', 'command': 'bash', 'run_args': '-t --rm'}, 'templates': {}, 'drone': {}, 'travis': {'enabled': False}, 'circle': {'enabled': False}, 'appveyor': {'image': 'Visual Studio 2017', 'enabled': False}, 'azure': {'settings_linux': {'pool': {'vmImage': 'ubuntu-16.04'}, 'timeoutInMinutes': 360, 'strategy': {'maxParallel': 8}}, 'settings_osx': {'pool': {'vmImage': 'macOS-10.14'}, 'timeoutInMinutes': 360, 'strategy': {'maxParallel': 8}}, 'settings_win': {'pool': {'vmImage': 'vs2017-win2016'}, 'timeoutInMinutes': 360, 'strategy': {'maxParallel': 4}, 'variables': {'CONDA_BLD_PATH': 'D:\\\\bld\\\\'}}, 'upload_packages': False, 'force': False, 'project_name': 'nsls2forge', 'project_id': '84710dde-1620-425b-80d0-4cf5baca359d', 'timeout_minutes': None, 'user_or_org': 'nsls2forge', 'setings_linux': {'pool': {'name': 'BNL-machines', 'demands': ['gpu -equals true']}, 'workspace': {'clean': 'all'}, 'strategy': {'maxParallel': 1}}, 'enabled': True, 'platforms': 'Linux', 'all_platforms': ['linux']}, 'provider': {'linux': 'azure', 'osx': 'azure', 'win': 'azure', 'linux_aarch64': None, 'linux_ppc64le': None, 'linux_armv7l': None}, 'win': {'enabled': False}, 'osx': {'enabled': False}, 'linux': {'enabled': True}, 'linux_aarch64': {'enabled': False}, 'linux_ppc64le': {'enabled': False}, 'linux_armv7l': {'enabled': False}, 'idle_timeout_minutes': None, 'compiler_stack': 'comp7', 'min_py_ver': '27', 'max_py_ver': '37', 'min_r_ver': '34', 'max_r_ver': '34', 'channels': {'sources': ['nsls2forge', 'defaults'], 'targets': [['nsls2forge', 'main']]}, 'github': {'user_or_org': 'nsls-ii-forge', 'repo_name': 'cupy-feedstock', 'branch_name': 'master'}, 'recipe_dir': 'recipe', 'skip_render': [], 'bot': {'automerge': False}, 'conda_forge_output_validation': False, 'upload_on_branch': 'master', 'package': 'cupy-feedstock', 'exclusive_config_file': '/home/leofang/nsls-ii-forge/cupy-feedstock/recipe/conda_build_config.yaml', 'feedstock_name': 'cupy-feedstock', 'migration_fns': [], 'configs': [{'config_name': 'linux_cuda_compiler_version10.0', 'platform': 'linux-64', 'upload': True, 'config': {'cuda_compiler_version': ['10.0'], 'docker_image': ['condaforge/linux-anvil-cuda:10.0'], 'nccl': ['2.4.6.1'], 'python': ['3.7'], 'cudnn': ['7.6.5'], 'channel_targets': ['nsls2forge main'], 'cxx_compiler': ['gxx'], 'c_compiler': ['gcc'], 'channel_sources': ['nsls2forge,defaults'], 'cuda_compiler': ['nvcc'], 'pin_run_as_build': OrderedDict([('cudnn', OrderedDict([('min_pin', 'x.x'), ('max_pin', 'x')])), ('python', OrderedDict([('min_pin', 'x.x'), ('max_pin', 'x.x')]))]), 'zip_keys': [['cuda_compiler_version', 'docker_image']]}, 'short_config_name': 'linux_cuda_compiler_version10.0'}, {'config_name': 'linux_cuda_compiler_version10.1', 'platform': 'linux-64', 'upload': True, 'config': {'cuda_compiler_version': ['10.1'], 'docker_image': ['condaforge/linux-anvil-cuda:10.1'], 'nccl': ['2.4.6.1'], 'python': ['3.7'], 'cudnn': ['7.6.5'], 'channel_targets': ['nsls2forge main'], 'cxx_compiler': ['gxx'], 'c_compiler': ['gcc'], 'channel_sources': ['nsls2forge,defaults'], 'cuda_compiler': ['nvcc'], 'pin_run_as_build': OrderedDict([('cudnn', OrderedDict([('min_pin', 'x.x'), ('max_pin', 'x')])), ('python', OrderedDict([('min_pin', 'x.x'), ('max_pin', 'x.x')]))]), 'zip_keys': [['cuda_compiler_version', 'docker_image']]}, 'short_config_name': 'linux_cuda_compiler_version10.1'}, {'config_name': 'linux_cuda_compiler_version10.2', 'platform': 'linux-64', 'upload': True, 'config': {'cuda_compiler_version': ['10.2'], 'docker_image': ['condaforge/linux-anvil-cuda:10.2'], 'nccl': ['2.4.6.1'], 'python': ['3.7'], 'cudnn': ['7.6.5'], 'channel_targets': ['nsls2forge main'], 'cxx_compiler': ['gxx'], 'c_compiler': ['gcc'], 'channel_sources': ['nsls2forge,defaults'], 'cuda_compiler': ['nvcc'], 'pin_run_as_build': OrderedDict([('cudnn', OrderedDict([('min_pin', 'x.x'), ('max_pin', 'x')])), ('python', OrderedDict([('min_pin', 'x.x'), ('max_pin', 'x.x')]))]), 'zip_keys': [['cuda_compiler_version', 'docker_image']]}, 'short_config_name': 'linux_cuda_compiler_version10.2'}, {'config_name': 'linux_cuda_compiler_version9.2', 'platform': 'linux-64', 'upload': True, 'config': {'cuda_compiler_version': ['9.2'], 'docker_image': ['condaforge/linux-anvil-cuda:9.2'], 'nccl': ['2.4.6.1'], 'python': ['3.7'], 'cudnn': ['7.6.5'], 'channel_targets': ['nsls2forge main'], 'cxx_compiler': ['gxx'], 'c_compiler': ['gcc'], 'channel_sources': ['nsls2forge,defaults'], 'cuda_compiler': ['nvcc'], 'pin_run_as_build': OrderedDict([('cudnn', OrderedDict([('min_pin', 'x.x'), ('max_pin', 'x')])), ('python', OrderedDict([('min_pin', 'x.x'), ('max_pin', 'x.x')]))]), 'zip_keys': [['cuda_compiler_version', 'docker_image']]}, 'short_config_name': 'linux_cuda_compiler_version9.2'}], 'fast_finish': '', 'upload_script': 'upload_or_check_non_existence', 'local_ci_setup': False}
```